samples/fast_with_profiling.exs

```diff
@@ -0,0 +1,157 @@
+# :erlang.apply/2                                     1  0.03    1    1.00
+# :erlang.++/2                                    10000 14.58  551    0.06
+# Enum.flat_map_list/2                            10001 42.13 1592    0.16
+# anonymous fn/1 in :elixir_compiler_2.__FILE__/1 10000 43.27 1635    0.16
+
+# Profile done over 6 matching functions
+
+# Profiling map.flatten with eprof...
+
+# Profile results of #PID<0.1619.0>
+# #                                               CALLS     % TIME µS/CALL
+# Total                                           60007 100.0 5281    0.09
+# List.flatten/1                                      1  0.00    0    0.00
+# Enum.map/2                                          1  0.00    0    0.00
+# :lists.flatten/1                                    1  0.00    0    0.00
+# :erlang.apply/2                                     1  0.04    2    2.00
+# anonymous fn/2 in :elixir_compiler_2.__FILE__/1     1  0.04    2    2.00
+# anonymous fn/1 in :elixir_compiler_2.__FILE__/1 10000 19.90 1051    0.11
+# Enum."-map/2-lists^map/1-1-"/2                  10001 26.60 1405    0.14
+# :lists.do_flatten/2                             40001 53.42 2821    0.07
+
+# Profile done over 8 matching functions
+# Operating System: Linux
+# CPU Information: AMD Ryzen 9 5900X 12-Core Processor
+# Number of Available Cores: 24
+# Available memory: 31.26 GB
+# Elixir 1.17.3
+# Erlang 27.1
+# JIT enabled: true
+
+# Benchmark suite executing with the following configuration:
+# warmup: 100 ms
+# time: 300 ms
+# memory time: 300 ms
+# reduction time: 0 ns
+# parallel: 1
+# inputs: none specified
+# Estimated total run time: 1 s 400 ms
+
+# Benchmarking flat_map ...
+# Benchmarking map.flatten ...
+# Calculating statistics...
+# Formatting results...
+
+# Name                  ips        average  deviation         median         99th %
+# flat_map           2.42 K      412.87 μs    ±48.32%      323.59 μs      737.84 μs
+# map.flatten        1.72 K      582.68 μs    ±35.48%      673.13 μs      987.99 μs
+
+# Comparison:
+# flat_map           2.42 K
+# map.flatten        1.72 K - 1.41x slower +169.81 μs
+
+# Memory usage statistics:
+
+# Name           Memory usage
+# flat_map             625 KB
+# map.flatten       781.25 KB - 1.25x memory usage +156.25 KB
+
+# **All measurements for memory usage were the same**
+
+# Profiling flat_map with tprof...
+
+# Profile results of #PID<0.3062.0>
+# #                                               CALLS      % TIME µS/CALL
+# Total                                           30003 100.00 3686    0.12
+# Enum.flat_map/2                                     1   0.00    0    0.00
+# anonymous fn/2 in :elixir_compiler_2.__FILE__/1     1   0.05    2    2.00
+# :erlang.++/2                                    10000  14.46  533    0.05
+# anonymous fn/1 in :elixir_compiler_2.__FILE__/1 10000  33.67 1241    0.12
+# Enum.flat_map_list/2                            10001  51.82 1910    0.19
```

Comment:
I wonder if there is a way to pass opts to the profilers, like `call_memory` to `tprof` to get heap allocs?

Reply:
`{profiler, opts}`

🤦

Reply:
Absolutely! See https://github.com/bencheeorg/benchee?tab=readme-ov-file#configuration for `profile_after`

> a profiler - either as a tuple of {profiler, opts} (e.g., {:fprof, [sort: :own]}) or just the profiler (e.g., :fprof), which is equivalent to {profiler, []}. The accepted built-in profilers are [:cprof](https://hexdocs.pm/mix/Mix.Tasks.Profile.Cprof.html), [:eprof](https://hexdocs.pm/mix/Mix.Tasks.Profile.Eprof.html) and [:fprof](https://hexdocs.pm/mix/Mix.Tasks.Profile.Fprof.html).

Should make that clearer/include it in the profiler section of the README though!

Reply:
ah didn't see your response, sometimes github auto fetches sometimes it doesn't - but yeah :)